curriculum/challenges/english/25-front-end-development/quiz-testing/66f1aeffc5774ac692112a7e.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
+End-to-end testing
 
 ---
 
-Placeholder distractor 3
+Integration testing
 
 #### --answer--
 
-Placeholder answer
+Unit testing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which testing approach evaluates a system's compliance with specified requirements from an end-user perspective?
 
 #### --distractors--
 
-Placeholder distractor 1
+Unit testing
 
 ---
 
-Placeholder distractor 2
+Performance testing
 
 ---
 
-Placeholder distractor 3
+Security testing
 
 #### --answer--
 
-Placeholder answer
+Functional testing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What type of testing assesses how well a system performs under various workloads?
 
 #### --distractors--
 
-Placeholder distractor 1
+Compatibility testing
 
 ---
 
-Placeholder distractor 2
+Usability testing
 
 ---
 
-Placeholder distractor 3
+Security testing
 
 #### --answer--
 
-Placeholder answer
+Performance testing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which testing practice involves evaluating a system's ability to protect data and maintain functionality in the face of malicious attacks?
 
 #### --distractors--
 
-Placeholder distractor 1
+Usability testing
 
 ---
 
-Placeholder distractor 2
+Compatibility testing
 
 ---
 
-Placeholder distractor 3
+Performance testing
 
 #### --answer--
 
-Placeholder answer
+Security testing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What type of testing ensures that a system can be used effectively by people with disabilities?
 
 #### --distractors--
 
-Placeholder distractor 1
+Compatibility testing
 
 ---
 
-Placeholder distractor 2
+Performance testing
 
 ---
 
-Placeholder distractor 3
+Functional testing
 
 #### --answer--
 
-Placeholder answer
+Accessibility testing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which popular JavaScript testing framework is often used for unit testing React applications?
 
 #### --distractors--
 
-Placeholder distractor 1
+Selenium
 
 ---
 
-Placeholder distractor 2
+Cypress
 
 ---
 
-Placeholder distractor 3
+Playwright
 
 #### --answer--
 
-Placeholder answer
+Jest
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What term describes the practice of testing an application across different web browsers to ensure consistent functionality?
 
 #### --distractors--
 
-Placeholder distractor 1
+Multi-platform testing
 
 ---
 
-Placeholder distractor 2
+Compatibility testing
 
 ---
 
-Placeholder distractor 3
+Integration testing
 
 #### --answer--
 
-Placeholder answer
+Cross-browser testing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which CI/CD practice involves automatically running tests whenever code changes are pushed to a repository?
 
 #### --distractors--
 
-Placeholder distractor 1
+Manual testing
 
 ---
 
-Placeholder distractor 2
+Regression testing
 
 ---
 
-Placeholder distractor 3
+Load testing
 
 #### --answer--
 
-Placeholder answer
+Automated tests on Github
```

Comment:
This question is bugging me for two reasons. Regression testing does seem like a valid answer since it is running tests to ensure code changes did not break existing functionality. 

The second reason is that it's technically automated testing being ran in GitHub Actions workflow, but the prompt did specifically say "Automated tests on Github". 

Source for Regression testing : https://www.geeksforgeeks.org/software-engineering-regression-testing/ 

Source on building a Github Actions workflow for testing: https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-net

Reply:
I agree question feels a bit off. It ask about _practice involving automatically running tests_ and answer is _Automated tests_.

Reply:
About the first concern, the question is specifically asking about the practice of automatically running tests when code changes are pushed, which is a broader concept than just regression testing as for the second one I agree that the answer provided is very specific.  I think switching the answer to Continuous Integration or just 'Automated tests'.

Or another thing we can do is change the question a bit to "Which CI/CD practice involves automatically running tests whenever code changes are pushed to a repository?"

Please lemme know which would be the better option.

Reply:
Let's see. 

"Which CI/CD practice involves automatically running tests whenever code changes are pushed to a repository?" 

A proper answer might be something like "Continuous Testing". 

Source: https://aws.amazon.com/what-is/continuous-testing/